### PR TITLE
stern: Include bash & zsh completion support

### PR DESCRIPTION
**Pull request by benley**
###### Motivation for this change
I want tab completion

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by matthewbauer** [comment]
@grahamcofborg eval

**Comment by roberth** [comment]
Sadly, it does break cross-compilation. I see two ways to fix it, either
 - don't generate completions when cross compiling
 - hardcode the completion scripts and turn the non-cross build into an equality test

The latter may or may not be feasible, depending on what's in the script. If the script defers most/all of the logic to the binary it should be fine.

File: pkgs/applications/networking/cluster/stern/default.nix [hunk 1]
```diff
@@ -15,11 +15,18 @@ buildGoPackage rec {
...
   goDeps = ./deps.nix;
 
-   meta = with lib; {
-      description      = "Multi pod and container log tailing for Kubernetes";
-      homepage         = "https://github.com/wercker/stern";
-      license          = licenses.asl20;
-      maintainers      = with maintainers; [ mbode ];
-      platforms        = platforms.unix;
-    };
+  postInstall = ''
```
Comment (Mic92) [inline pkgs/applications/networking/cluster/stern/default.nix]: In the cross compiling case we could use the the output of `buildPackages`.
However the fix for the meantime would be also:

```
postInstall = optionalString (stdenv.buildPlatform != stdenv.hostPlatform) ''
  mkdir -p $bin/share/bash-completion/completions
  $bin/bin/stern --completion bash > $bin/share/bash-completion/completions/stern
  mkdir -p $bin/share/zsh/site-functions
  $bin/bin/stern --completion zsh > $bin/share/zsh/site-functions/_stern
'';
```

Comment (benley) [reply pkgs/applications/networking/cluster/stern/default.nix]: Oh nice, that's what I should have done 23 days ago. Coming right up.

**Pull request merged**